python/sglang/srt/layers/attention/nsa/nsa_indexer.py

```diff
@@ -261,6 +261,88 @@ def _get_q_k_bf16(
 
         return query, key, weights
 
+    def _construct_attention_indices(
```

Comment:
I think it's way easier to construct an empty matrix the same size as the logits matrix [here](https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/layers/attention/nsa/nsa_indexer.py#L361) and pass that to the topk kernel so it can generate the indices for you. It calls one kernel instead of the multiples kernels down below. The topk kernel generates the indices directly when the length is <= 2k. See https://github.com/sgl-project/sglang/blob/34f7564df0378d2dba929549c0350cbc12713666/sgl-kernel/csrc/elementwise/topk.cu#L35-L40